.husky/pre-commit

```diff
@@ -2,4 +2,4 @@
 . "$(dirname "$0")/_/husky.sh"
 
 yarn build
-git add lib
+git add packages/near-api-js/lib
```

Comment:
This is fine for now but may become a problem as we add more packages, i.e. if we update a package other than `near-api-js` we don't really want to be running this commit hook. Especially since `yarn build` will build every package, it could become time consuming. 

Is there a way to scope this just to `near-api-js`?

Reply:
I think the idea is to eventually generate the lib on merge or on publish right? Ideally we don't want these files in every diff. Let's sync on this 🙂

Reply:
After our sync and researching, I think we can use [the `prepare` script](https://docs.npmjs.com/cli/v8/using-npm/scripts#life-cycle-scripts) for this purpose. We can remove and gitignore the `lib` directory that way and also the `prepublish` script. The source for the `not-in-install` util currently being used in pre publish is the [`in-publish` npm package](https://www.npmjs.com/package/in-publish). The main purpose of this is to ensure the scripts aren't run on npm install which I think isn't necessary for our purposes so removing that too.

cc: @volovyk-s

Reply:
If you are moving to lerna then you can no longer use git as a dependency so I think it's best to remove it altogether.

Reply:
I was thinking about that but looks like it can be added as a sub directory? see https://gitpkg.vercel.app/

Reply:
Nice! I wish I had that when making the near-sdk-as repo!